quinn/src/platform/unix.rs

```diff
@@ -290,12 +290,40 @@ fn prepare_msg(
     hdr.msg_control = ctrl.0.as_mut_ptr() as _;
     hdr.msg_controllen = CMSG_LEN as _;
     let mut encoder = unsafe { cmsg::Encoder::new(hdr) };
+
     let ecn = transmit.ecn.map_or(0, |x| x as libc::c_int);
     if transmit.destination.is_ipv4() {
         encoder.push(libc::IPPROTO_IP, libc::IP_TOS, ecn as IpTosTy);
     } else {
         encoder.push(libc::IPPROTO_IPV6, libc::IPV6_TCLASS, ecn);
     }
+
+    if let Some(ip) = &transmit.dst_ip {
+        if cfg!(target_os = "linux") {
+            match ip {
+                IpAddr::V4(v4) => {
+                    let pktinfo = libc::in_pktinfo {
+                        ipi_ifindex: 0,
+                        ipi_spec_dst: libc::in_addr { s_addr: 0 },
+                        ipi_addr: unsafe {
+                            *(v4 as *const Ipv4Addr as *const () as *const libc::in_addr)
```

Comment:
Are we allowed to make this cast? I thought this was being reconsidered.

https://github.com/rust-lang/rust/pull/78802

Reply:
It definitely is based on assumptions of the stdlib. However those are already made in a few other places. I think if we are not comfortable with them, all those should be updated.

Reply:
Fair enough, filed https://github.com/quinn-rs/quinn/issues/968 about this.